porcupine/plugins/rightclick_menu.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import annotations
+
+from typing import Callable, Any
+from porcupine import get_main_window, tabs
+from porcupine.menubar import get_filetab
+
+import tkinter
+
+_ftab_indexes_ = []
+_menu_items_ = {}
+
+
+def text_is_selected(tab: tabs.FileTab) -> bool:
+    try:
+        tab.textwidget.get("sel.first", "sel.last")
+    except tkinter.TclError:
+        # nothing selected
+        return False
+    return True
+
+
+def show_menu(event) -> None:
+    rm = tkinter.Menu(get_main_window(), tearoff=0)
+
+    for path, func in _menu_items_.items():
+        rm.add_command(label=path, command=func)
+
+
+    if rm.index("end") is not None:
+        flag = bool
+        try:
+            flag=text_is_selected(get_filetab())
+        except AssertionError:
+            flag=False
+        finally:
+            if flag:
+                for i in _ftab_indexes_:
+                    rm.entryconfigure(i, state=tkinter.NORMAL)
+            else:
+                for i in _ftab_indexes_:
+                    rm.entryconfigure(i, state=tkinter.DISABLED)
+        rm.tk_popup(event.x_root, event.y_root)
+        rm.bind("<Unmap>", (lambda event: rm.after_idle(rm.update)), add=True)
+
+def add_rightclick_option(path: str, func):
```

Comment:
Maybe the same function could also handle menu items that are only for selected text? It could be an optional argument. Like this, for example:
```suggestion
def add_rightclick_option(path: str, func, disable_when_nothing_selected: bool = False):
```
Then we wouldn't need the `add_filetab_command()` function below. To be honest, it's really quite confusing and I would like to get rid of it :D But I like the `add_rightclick_option()` function.